src/filters/blur_filter.class.ts

```diff
@@ -65,14 +64,14 @@ export class Blur extends BaseFilter {
       resources.blurLayer1 = createCanvasElement();
       resources.blurLayer2 = createCanvasElement();
     }
-    const canvas1 = resources.blurLayer1;
-    const canvas2 = resources.blurLayer2;
+    const canvas1 = resources.blurLayer1!;
+    const canvas2 = resources.blurLayer2!;
     if (canvas1.width !== width || canvas1.height !== height) {
       canvas2.width = canvas1.width = width;
       canvas2.height = canvas1.height = height;
     }
-    const ctx1 = canvas1.getContext('2d'),
-      ctx2 = canvas2.getContext('2d'),
+    const ctx1 = canvas1.getContext('2d')!,
+      ctx2 = canvas2.getContext('2d')!,
```

Comment:
we probably need a type that is HTMLCanvasElementThatWorks that never return null on getContext('2d').
We just need to agree that if your machine has no memory to create canvases anymore fabricJS is not going to work anyway.

Reply:
We can modify the global type of the window/dom

Reply:
Or better off,
we use `AssertKeys` in `createCanvasElement`